Tests/RxCocoaTests/UITabBarController+RxTests.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  UITabBarController+RxTests.swift
+//  Tests
+//
+//  Created by Yusuke Kita on 2016/12/07.
+//  Copyright © 2016 Krunoslav Zaher. All rights reserved.
+//
+
+#if os(iOS) || os(tvOS)
+
+import Foundation
+
+import RxSwift
+import RxCocoa
+import UIKit
+import XCTest
+
+class UITabBarControllerTests: RxTest {
+    let createSubject: () -> UITabBarController = { UITabBarController() }
+}
+
+/**
+ iOS only
+ */
+#if os(iOS)
+extension UITabBarControllerTests {
+    func testWillBeginCustomizing() {
+        let subject = createSubject()
+        
+        let viewControllers = [UIViewController()]
+        var returnedViewControllers = [UIViewController]()
+        
+        XCTAssertNotEqual(returnedViewControllers, viewControllers)
+        
+        _ = subject.rx.willBeginCustomizing
+            .subscribe(onNext: { vc in
+                returnedViewControllers = vc
+            })
+        
+        subject.delegate!.tabBarController!(subject, willBeginCustomizing: viewControllers)
+        
+        XCTAssertEqual(returnedViewControllers, viewControllers)
+    }
+    
+    func testWillEndCustomizing() {
+        let subject = createSubject()
+        
+        let viewControllers = [UIViewController()]
+        var returnedViewControllers = [UIViewController]()
+        var changed: Bool!
+        
+        XCTAssertNotEqual(returnedViewControllers, viewControllers)
+        XCTAssertNotEqual(changed, true)
```

Comment:
These methods should be after `subject.rx.willEndCustomizing` because we are testing that extensions haven't emitted anything initially.